### PR TITLE
Port Magic Crayon From Frontier

### PR DESCRIPTION
**Pull request by Avalon-Proto**
## About the PR
Ported the rainbow crayon from Frontier to here! Artists Rejoice!
## Why / Balance
We have a number of crayon artists on Delta ~~cough Joy Bringer cough~~ so i wanted to give them something they can really use for art

## Media
https://cdn.discordapp.com/attachments/1206353544186171482/1315816832048828477/image.png?ex=6758c96b&is=675777eb&hm=30f78de869bf562a16551bc066d17a6d1f939f800abb6972cea293fd73516097&

## Requirements

- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
🆑 add: New magic crayons have appeared in the sector, artists rejoice!

### CODE REVIEW

**Comment by Avalon-Proto** [comment]
I jsut need help getting it into loadouts, as I currently cannot figure out how to

**Comment by Lyndomen** [comment]
> I jsut need help getting it into loadouts, as I currently cannot figure out how to

check my sec gun PR

**Comment by Lyndomen** [comment]
Also pending Direction review, but I have some concerns providing this at roundstart with infinite uses. I'd much rather this be construct-able with real effort, researchable, or really anything but free.

**Comment by Kr8art** [comment]
> Also pending Direction review, but I have some concerns providing this at roundstart with infinite uses. I'd much rather this be construct-able with real effort, researchable, or really anything but free.

If this doesn't get approved, I might do a PR for a rainbow crayon crafting recipe that lets you combine all crayons from a box to make one rainbow crayon.

That way artists still have a source of crayons that can actually paint all colours, but they'd have to do some work for it (finding them/asking logi for a few boxes).

**Comment by Avalon-Proto** [comment]
> Also pending Direction review, but I have some concerns providing this at roundstart with infinite uses. I'd much rather this be construct-able with real effort, researchable, or really anything but free.

I mean, I plan to lock it behind at least 5 hours playtime of each role, as well as giving it a small chance to appear in artists toolboxes. About every 1 in 3 boxes can have one. Or 1-100 of getting one from a maints locker on its own

**Comment by Avalon-Proto** [comment]
Right, now locked behind 10 hours clown, for clown and mime. Cause I could NOT figure out how to make it require both

**Comment by Lyndomen** [comment]
Alrighty, so the direction review is that we **do not want this available roundstart via loudouts.** Our suggestion is that it becomes craftable through 1 red, green, and blue crayon, as well as a bluespace crystal.

**Comment by Avalon-Proto** [comment]
> Alrighty, so the direction review is that we **do not want this available roundstart via loudouts.** Our suggestion is that it becomes craftable through 1 red, green, and blue crayon, as well as a bluespace crystal.

Alright help me do the construction graph then cause I dunno how to make those :blunt: . But is the 33% chance to spawn in a green toolbox. And 1/100 for Maints lockers fine?

**Comment by Lyndomen** [comment]
You can check the Frontier port of Coffee to see construction graphs, and the 1/100 is fine but please change the green toolbox chance down to 5%

**Comment by Avalon-Proto** [comment]
> You can check the Frontier port of Coffee to see construction graphs, and the 1/100 is fine but please change the green toolbox chance down to 5%

I mean the only reason I made it 33% in green boxes is cause green boxes themselves are relatively rare. I tested opening around 30 Maints lockers. And in those lockers I got 0 magic crayons and 2 art boxes. Neither art box had a crayon

File: Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml [hunk 1]
```diff
@@ -85,6 +85,8 @@
       amount: 2 
     - id: CrayonMime
     - id: CrayonRainbow
+    - id: CrayonMagic # Delta V - I don't feel liek making a construction graph :blunt:
+      prob: .33
```
Comment (Lyndomen) [inline Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml]: .05, .1 tops

File: Resources/Prototypes/DeltaV/Loadouts/Jobs/Civilian/clown.yml [hunk 2]
```diff
@@ -13,3 +13,16 @@
   id: FoolPDA
   equipment:
     id: FoolPDA
+
+# Trinkets
+- type: loadout
```
Comment (Lyndomen) [inline Resources/Prototypes/DeltaV/Loadouts/Jobs/Civilian/clown.yml]: remove

File: Resources/Prototypes/DeltaV/Loadouts/Jobs/Civilian/mime.yml [hunk 3]
```diff
@@ -3,3 +3,16 @@
   id: FoamSabre
   equipment:
     belt: ClothingBeltFoamSheathFilled
+
+# Trinkets
```
Comment (Lyndomen) [inline Resources/Prototypes/DeltaV/Loadouts/Jobs/Civilian/mime.yml]: remove

File: Resources/Prototypes/Loadouts/loadout_groups.yml [hunk 4]
```diff
@@ -51,6 +51,7 @@
   - Cane # DeltaV
   - WhiteCane #DeltaV
   - CDDogtags # _CD
+  - CrayonMagic # Delta V, _NF Port
```
Comment (Lyndomen) [inline Resources/Prototypes/Loadouts/loadout_groups.yml]: remove

File: SpaceStation14.sln [hunk 5]
```diff
@@ -471,21 +471,21 @@ Global
 		{41B450C0-A361-4CD7-8121-7072B8995CFC} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{7B9472D3-79D4-48D1-9B22-BCDE518FE842} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{1FAE651D-29D8-437A-9864-47CE0D180016} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
+		{3CFEB7DB-12C6-46F3-89FC-1450F3016FFA} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
```
Comment (Lyndomen) [inline SpaceStation14.sln]: im not a maints but untouch probably

File: SpaceStation14.sln [hunk 6]
```diff
@@ -471,21 +471,21 @@ Global
...
 		{8922428F-17C3-47A7-BFE9-570DEB2464DA} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{16F7DE32-0186-44B9-9345-0C20D1BF2422} = {AFF53804-115F-4E67-B81F-26265EA27880}
 		{AFF53804-115F-4E67-B81F-26265EA27880} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{23F09C45-950E-4DB7-A465-E937450FF008} = {AFF53804-115F-4E67-B81F-26265EA27880}
 		{440426C1-8DCA-43F6-967F-94439B8DAF47} = {AFF53804-115F-4E67-B81F-26265EA27880}
+		{88B0FC0F-7209-40E2-AF16-EB90AF727C5B} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
```
Comment (Lyndomen) [inline SpaceStation14.sln]: im not a maints but untouch probably

File: SpaceStation14.sln [hunk 7]
```diff
@@ -471,21 +471,21 @@ Global
...
 		{A3C5B00A-D232-4A01-B82E-B0E58BFD5C12} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{8A21C7CA-2EB8-40E5-8043-33582C06D139} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{952AAF2A-DF63-4A7D-8094-3453893EBA80} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
 		{A965CB3B-FD31-44AF-8872-85ABA436098D} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
-		{7844DA69-B0F0-49FB-A05E-ECA37372277A} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
-		{3CFEB7DB-12C6-46F3-89FC-1450F3016FFA} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
-		{6FBF108E-5CB5-47DE-8D7E-B496ABA9E3E2} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
 		{07CA34A1-1D37-4771-A2E3-495A1044AE0B} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
-		{88B0FC0F-7209-40E2-AF16-EB90AF727C5B} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
+		{6FBF108E-5CB5-47DE-8D7E-B496ABA9E3E2} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
```
Comment (Lyndomen) [inline SpaceStation14.sln]: im not a maints but untouch probably

File: SpaceStation14.sln [hunk 8]
```diff
@@ -471,21 +471,21 @@ Global
...
 		{83F510FE-9B50-4D96-AFAB-CC13998D6AFE} = {7844DA69-B0F0-49FB-A05E-ECA37372277A}
+		{7844DA69-B0F0-49FB-A05E-ECA37372277A} = {83B4CBBA-547A-42F0-A7CD-8A67D93196CE}
```
Comment (Lyndomen) [inline SpaceStation14.sln]: im not a maints but untouch probably

File: BuildChecker/BuildChecker.csproj [hunk 9]
```diff
@@ -19,6 +19,8 @@ https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild
     <ProjectGuid>{C899FCA4-7037-4E49-ABC2-44DE72487110}</ProjectGuid>
     <TargetFrameworkMoniker>.NETFramework, Version=v4.7.2</TargetFrameworkMoniker>
     <RestorePackages>false</RestorePackages>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
```
Comment (Lyndomen) [inline BuildChecker/BuildChecker.csproj]: im not a maints but untouch probably

File: BuildChecker/BuildChecker.csproj [hunk 10]
```diff
@@ -19,6 +19,8 @@ https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild
...
+    <TargetFrameworkProfile />
```
Comment (Lyndomen) [inline BuildChecker/BuildChecker.csproj]: im not a maints but untouch probably

File: BuildChecker/BuildChecker.csproj [hunk 11]
```diff
@@ -49,4 +51,4 @@ https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild
   </Target>
   <Target Name="CoreCompile">
   </Target>
-</Project>
+</Project>
```
Comment (Lyndomen) [inline BuildChecker/BuildChecker.csproj]: im not a maints but untouch probably

Comment (Avalon-Proto) [reply BuildChecker/BuildChecker.csproj]: Imma be honest I dunno what it means to untouch

File: Resources/Prototypes/Loadouts/loadout_groups.yml (hunk 4 above)
Comment (Avalon-Proto) [reply Resources/Prototypes/Loadouts/loadout_groups.yml]: Will be rmeoved, jsut havent pushed my latest commit

File: BuildChecker/BuildChecker.csproj (hunk 11 above)
Comment (Radezolid) [reply BuildChecker/BuildChecker.csproj]: it means to remove the change you did there.

Comment (Avalon-Proto) [reply BuildChecker/BuildChecker.csproj]: > it means to remove the change you did there.

I never even touched that file is why I'm confused :blunt:

**Review by Lyndomen — DISMISSED** [review]
no construction but rare rng ok

File: Resources/Prototypes/_NF/Entities/Objects/Fun/magic_crayon.yml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  - type: FlavorProfile
+    flavors:
+    - chewy
+    - magical
+  - type: SolutionContainerManager
+    solutions:
+      food:
+        reagents:
+        - ReagentId: Nothing
+          Quantity: 40
```
Comment (MilonPL) [inline Resources/Prototypes/_NF/Entities/Objects/Fun/magic_crayon.yml]: EOF newline

File: Resources/Locale/en-US/crayon/crayon-component.ftl [hunk 13]
```diff
@@ -5,6 +5,7 @@ crayon-drawing-label = Drawing: [color={$color}]{$state}[/color] ({$charges}/{$c
 crayon-interact-not-enough-left-text = Not enough left.
 crayon-interact-used-up-text = The {$owner} got used up.
 crayon-interact-invalid-location = Can't reach there!
+crayon-drawing-label-unlimited = Drawing: [color={$color}]{$state}[/color]
```
Comment (MilonPL) [inline Resources/Locale/en-US/crayon/crayon-component.ftl]: this shouldn't be in upstream namespace

File: BuildChecker/BuildChecker.csproj
Comment (MilonPL) [inline BuildChecker/BuildChecker.csproj:1]: checkout unrelated files, idk why you even committed changes to them

File: SpaceStation14.sln
Comment (MilonPL) [inline SpaceStation14.sln:1]: .

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Comment by Avalon-Proto** [comment]
> no construction but rare rng ok

I have a construction graph Radez there gave me that I will be testing once I am home. And if it works I'll be adding it to the PR too

File: Resources/Prototypes/_NF/Entities/Objects/Fun/magic_crayon.yml (hunk 12 above)
Comment (Avalon-Proto) [reply Resources/Prototypes/_NF/Entities/Objects/Fun/magic_crayon.yml]: Fixing this, also making that "Nothing" reagent 100 so it takes 20 bites to eat, and people dont accidentally eat their rare, magic cray. Since nothing does nothing, should be fine

**Comment by Avalon-Proto** [comment]
Okay, so I have everything working but the construction graph, I moved the crayon component.. but now I need to figure out why the graph is not working :elp:

**Comment by Radezolid** [comment]
Did you add the construction to the magic crayon? [Example](https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml#L998)

**Comment by Avalon-Proto** [comment]
> Did you add the construction to the magic crayon? [Example](https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml#L998)

I have now, but its still not working

File: Resources/Locale/en-US/_NF/crayon/crayon-component.ftl [hunk 14]
```diff
@@ -0,0 +1,7 @@
+
+## Entity
+
+crayon-drawing-label-unlimited = Drawing: [color={$color}]{$state}[/color]
+
+## UI
+crayon-window-title = Crayon
```
Comment (Radezolid) [inline Resources/Locale/en-US/_NF/crayon/crayon-component.ftl]: This comes from upstream already, so it's duplicated. YAML Linter is screaming at this existing

File: Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml [hunk 15]
```diff
@@ -0,0 +1,28 @@
+- type: constructionGraph
+  id: CrayonMagic
```
Comment (Radezolid) [inline Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: Change to `magic_crayon` to match the crafting recipe.

File: Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+  start: start
+  graph:
+    - node: start
+      edges:
+        - to: magicCrayon
+          steps:
+            - material: MaterialBluespace1
```
Comment (Radezolid) [inline Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: Change to `Bluespace`

File: Resources/Locale/en-US/_NF/crayon/crayon-component.ftl (hunk 14 above)
Comment (Avalon-Proto) [reply Resources/Locale/en-US/_NF/crayon/crayon-component.ftl]: What do you suggest I rename it to? I need that so that the UI is not cluttered up w/ the ridiculous maxinteger

Comment (Radezolid) [reply Resources/Locale/en-US/_NF/crayon/crayon-component.ftl]: Nuke that line. That locale is already coming from the upstream `crayon-component.ftl`

File: Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml (hunk 15 above)
Comment (Avalon-Proto) [reply Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: Still not there

File: Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml (hunk 16 above)
Comment (Avalon-Proto) [reply Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: Fixed

File: Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml (hunk 15 above)
Comment (Radezolid) [reply Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: I didn't understand your comment.

Comment (Avalon-Proto) [reply Resources/Prototypes/_NF/Recipes/Crafting/Graphs/magic_crayon.yml]: Wait feck I looked at the wrong file

File: BuildChecker/BuildChecker.csproj
Comment (Avalon-Proto) [reply BuildChecker/BuildChecker.csproj:1]: When you say checkout, whaddya mean? Not sure I get the term. Can i just, undo any changes I might have made or

**Comment by Avalon-Proto** [comment]
Hey @deltanedas Can you tell me what i need to do to untouch stuff, cause I don't even remember opening those files :blunt:

File: BuildChecker/BuildChecker.csproj (hunk 11 above)
Comment (Radezolid) [reply BuildChecker/BuildChecker.csproj]: Giving a hand, i don't know git properly, what i do in these cases is do the inverse of the changes here. it seems it removed a end of line there, try doing a reset to it's previous commit.

**Comment by deltanedas** [comment]
`git checkout upstream/master path/to/file` where upstream is the name of your DeltaV-Station/Delta-v` remote

**Comment by Avalon-Proto** [comment]
> `git checkout upstream/master path/to/file` where upstream is the name of your DeltaV-Station/Delta-v` remote

Do I do this in smartgit, in Visual studio? Where?

**Comment by deltanedas** [comment]
in a terminal

**Comment by Avalon-Proto** [comment]
FINALLY!! FINALLY GOT IT THANK FOCK

**Comment by dvir001** [comment]
> FINALLY!! FINALLY GOT IT THANK FOCK

It do be like this.

**Comment by Avalon-Proto** [comment]
> > FINALLY!! FINALLY GOT IT THANK FOCK
> 
> It do be like this.

Untouching stuff is annoying when you don't know how to :elp:

File: Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml [hunk 17]
```diff
@@ -85,6 +85,8 @@
       amount: 2 
     - id: CrayonMime
     - id: CrayonRainbow
+    - id: CrayonMagic # Delta V - I don't feel like making a construction graph :blunt:
```
Comment (Radezolid) [inline Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml]: You got a construction graph now, maybe remove that part of the comment but keep the `# DeltaV`

Comment (Avalon-Proto) [reply Resources/Prototypes/Catalog/Fills/Items/toolboxes.yml]: ... yknow thats fair,

**Comment by Avalon-Proto** [comment]
@deltanedas I think this is all good now

File: Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    "version": 1,
+    "license": "CC-BY-SA-4.0",
+    "copyright": "Taken from tgstation and modified by Swept at commit https://github.com/tgstation/tgstation/commit/c6e3401f2e7e1e55c57060cdf956a98ef1fefc24, tweaked by Ubaser and whatston3 (GitHub)",
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "states": [
+        {
+        "name": "icon",
```
Comment (deltanedas) [inline Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json]: indent is fucked for this state

**Comment by Monotheonist** [comment]
Don't forget the mail, silly.

**Review by Lyndomen — DISMISSED** [review]
direction approved

**Comment by Avalon-Proto** [comment]
Added one last thing- gave it a 20% chance to come from the Mime Art Supply mail, and a 10% from the standard crayon mail

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json (hunk 18 above)
Comment (Avalon-Proto) [reply Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json]: Oh weird, which way do I nudge it? And sorry I missed this comment it got buried

Comment (deltanedas) [reply Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json]: indent right by 4 spaces

Comment (Avalon-Proto) [reply Resources/Textures/_NF/Objects/Fun/magic_crayon.rsi/meta.json]: Webedit ops fix :blunt:

**Review by deltanedas — APPROVED** [review]

**Comment by Avalon-Proto** [comment]
FUCK I FUCKED THA CHANGELOG
